torch/autograd/forward_ad.py

```diff
@@ -118,16 +117,32 @@ def make_dual(tensor, tangent, *, level=None):
             f"Expected tangent to be floating point or complex, but got: {tangent.dtype}"
         )
 
-    return torch._VF._make_dual(tensor, tangent, level=level)
+    return torch._VF._make_dual(tensor, tangent, level=level)  # type: ignore[attr-defined]
 
 
-_UnpackedDualTensor = namedtuple("_UnpackedDualTensor", ["primal", "tangent"])
-
-
-class UnpackedDualTensor(_UnpackedDualTensor):
+class UnpackedDualTensor(tuple):
```

Comment:
Now a subclass of a namedtuple class is also considered as a namedtuple class. So it will be treated as a pytree node. Here we make `UnpackedDualTensor` a custom class (do not use `namedtuple`) to let it still be a leaf type.

Reply:
What is the user-visible change related to this? How should we help user understand what we're changing here so they can adapt their code as needed.

Reply:
Users will not need to change their code. The new implementation still looks like a namedtuple and still remains a leaf type in our pytree utilities.

```python
# Create a new instance like before
unpacked = UnpackedDualTensor(primal, tangent)

# Tuple unpacking
primal, tangent = unpacked

# Named access
primal = unpacked.primal
tangent = unpacked.tangent
```